### ISSUE TITLE
How to make the sidebar more wider?

### ISSUE THREAD

**Issue by joshuarobs** [ankitskvmdam/clean-jsdoc-theme#50]
Currently its set at 250px by default, is there an option to set this to make it more wider? I tried looking into the css and changing the width myself, but it doesn't make the main content move, i.e. the sidebar gets wider but covers the docs

**Comment by ankitskvmdam** [comment]
There are two ways of doing this.
1. In your `jsdoc-config.json` under `opts` first you have to include `theme_opts` key. Then under `theme_opts` add `create_style` key with value, `"nav {width: 300px} #main {margin-left: 300px} @media (max-width: 672px) { nav {left: -300px} #main{margin-left: 0}}"`. Note: nav width must be equal to #main margin-left.
Your `jsdoc-config` file will look something like this:
```json5
// jsdoc-config.json
{ 
    ...
    ...
    "opts": {
        ...
        ...
        "theme_opts": {
            "create_style": "nav {width: 300px} #main {margin-left: 300px} @media (max-width: 672px) { nav {left: -300px} #main{margin-left: 0}}"
        }
    },
}
```

2. Alternatively you can add your custom css file path:
```css
/* custom.css */
nav {
    width: 300px
} 

#main {
    margin-left: 300px
}

@media (max-width: 672px) {
    nav {
        left: -300px
    } 

    #main {
        margin-left: 0
    }
}
```
```json5
// jsdoc-config.json
{ 
    ...
    ...
    "opts": {
        ...
        ...
        "theme_opts": {
           "add_style_path": ["custom.css"] //Note path must be relative to your output file. (relative to generated html files.)
           // And you have to manually copy all the css to the output directory.
        }
    },
}
```

**Comment by ankitskvmdam** [comment]
@joshuarobs due to no activity I am closing this issue, if you still facing a problem, feel free to open it again.

**Comment by kolban-google** [comment]
Can the side bar be made so that it is dynamically adjustable?  A window sash style?

**Comment by ankitskvmdam** [comment]
@kolban-google can you please throw more light on this?

**Comment by kolban-google** [comment]
When I look at the output of the JSDoc processing ... I see that there is a navigation area on the left and a content area on the right.  For the 3rd party JavaScript classes I am looking at, the names of the classes are so long that they get truncated in the navigation area on the left.  Ideally what I'd like to do is expand the width of the navigation area.   To my mind a "paned" window with a resized separator that can be dragged left or right to change the relative width of the navigation area would be ideal.

**Comment by ankitskvmdam** [comment]
Hmm... This is really an amazing idea. What we can do is, when the user is hovering near the border of the sidebar then the default cursor will change into the resize cursor, which allows the user to change the size of the sidebar. Also, I am thinking to have a default value for the min and max width of the sidebar which is configurable through jsdoc config.

Thanks, @kolban-google for this wonderful idea. I think we can expect this in the next release.

**Comment by ankitskvmdam** [comment]
Hi @kolban-google , I have added an option to adjust the navbar/sidebar dynamically. You can see the demo at https://ankdev.me/clean-jsdoc-theme/dark/index.html
Also, this feature is available in v3.2.1
For options related to resizeable is at https://github.com/ankitskvmdam/clean-jsdoc-theme/#resizeable-

**Comment by kolban-google** [comment]
Oh wow!!!  This looks FANTASTIC ... ABSOLUTELY FANTASTIC!!!  The docs look great and the demo looks YUMMY!   I'm going to start putting it through its paces.

For some back story, I am working a lot with Google Cloud products and their JSDoc is on the web at the following:

https://googleapis.dev/nodejs/dialogflow/latest/index.html

When I try and read these pages, I find myself frustrated with the truncation.  That's when I started looking at alternate Jsdoc templates and it looks like yours will be super yummy.  Especially now you have added the resizing!!

THANK YOU... Neil

**Comment by kolban-google** [comment]
You have a slight typo in the README.  I found it when copying pasting.  You have:

```
"resizeable: {
  "navbar": {
    "min": "300",
    "max": "600"
  }
}
```

which I believe should be:

```
"resizeable": {
  "navbar": {
    "min": "300",
    "max": "600"
  }
}
```

**Comment by ankitskvmdam** [comment]
@kolban-google I am glad you liked this project! And, Thanks for reporting the typos, I have fixed this as well.

**Comment by ankitskvmdam** [comment]
@kolban-google , It is just for a quick update, there is a bug in v3.2.2, link: [#68 ](https://github.com/ankitskvmdam/clean-jsdoc-theme/issues/68), this is fixed in v3.2.4
